File: src/upros_depth_vision/scripts/upros_image.py
#!/usr/bin/env python3

import rospy
from sensor_msgs.msg import Image
from cv_bridge import CvBridge
from cv_bridge import CvBridgeError
import cv2

def image_callback(msg):
    try:
        bridge = CvBridge()
        img = bridge.imgmsg_to_cv2(msg, "bgr8")
        cv2.imshow("Image", img)
        cv2.waitKey(1)
    except CvBridgeError as e:
        rospy.logerr("cv_bridge exception: %s" % e)

if __name__ == '__main__':
    rospy.init_node('image_subscriber')
    rospy.Subscriber('/camera/color/image_raw', Image, image_callback)
    cv2.namedWindow("Image", cv2.WINDOW_NORMAL)
    rospy.spin()
    cv2.destroyWindow("Image")

File: src/upros_depth_vision/src/upros_image.cpp
#include <ros/ros.h>
#include <sensor_msgs/Image.h>
#include <image_transport/image_transport.h>
#include <cv_bridge/cv_bridge.h>
#include <opencv2/opencv.hpp>

void imageCallback(const sensor_msgs::ImageConstPtr &msg)
{
    try
    {
        cv::Mat img;
        img = cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::BGR8)->image;
        // 在这里对图像进行处理
        cv::imshow("Image", img);
        cv::waitKey(1);
    }
    catch (cv_bridge::Exception &e)
    {
        ROS_ERROR("cv_bridge exception: %s", e.what());
    }
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "image_subscriber");
    ros::NodeHandle nh;
    image_transport::ImageTransport it(nh);
    image_transport::Subscriber sub = it.subscribe("/camera/color/image_raw", 1, imageCallback);
    cv::namedWindow("Image");
    ros::spin();
    cv::destroyWindow("Image");
    return 0;
}
